service/watcher/configmap/watch.go

```diff
@@ -0,0 +1,153 @@
+package configmap
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+
+	"github.com/giantswarm/app-operator/v2/pkg/annotation"
+	applabel "github.com/giantswarm/app-operator/v2/pkg/label"
+)
+
+func (c *AppValue) watch(ctx context.Context) {
+	for {
+		lo := metav1.ListOptions{
+			LabelSelector: applabel.Watching,
+		}
+
+		// Found the highest resourceVersion in cofigMaps CRs
```

Comment:
```suggestion
		// Find the highest resourceVersion for each configmap.
```

Some more comments on this part would be good. As the logic is quite complex.